node/pkg/watchers/aptos/watcher.go

```diff
@@ -100,17 +121,181 @@ func (e *Watcher) Run(ctx context.Context) error {
 		zap.String("healthQuery", aptosHealth),
 	)
 
-	// the events have sequence numbers associated with them in the aptos API
-	// (NOTE: this is not the same as the wormhole sequence id). The event
-	// endpoint is paginated, so we use this variable to keep track of which
-	// sequence number to look up next.
+	// Important: There are 3 different sequence/version numbers in play:
+	// 1. Aptos Version: Transaction version number (like block number in EVM)
+	// 2. Aptos Sequence Number: Event-specific counter for WormholeMessage events
+	// 3. Wormhole Sequence: Protocol sequence inside the event data (goes into VAA)
+	//
+	// nextSequence tracks the Aptos Sequence Number (not version, not wormhole seq)
 	var nextSequence uint64 = 0
```

Comment:
nit: This value is hard-coded to 0 and it's an argument to private methods. So it seems like it's not necessary to define here? The value could just be initialized within the methods themselves, and the parameter could be removed from the function signature.

Reply:
Changed